lingering ionKind in compiler unit tests borks the build

```
test/unit-modcc/test_parser.cpp
167:        EXPECT_EQ(s->ion_channel(), ionKind::Na);
173:        EXPECT_EQ(s->ion_channel(), ionKind::nonspecific);

test/unit/test_mechanisms.cpp
77:    std::map<ionKind, ion<typename T::backend>> ions;
```

leads to:

```
bcumming@arapiles:build > make unit-modcc
[  6%] Built target gtest
[ 60%] Built target libmodcc
[ 60%] Building CXX object test/unit-modcc/CMakeFiles/unit-modcc.dir/test_parser.cpp.o
In file included from /home/bcumming/software/github/arbor/test/unit-modcc/common.hpp:5,
                 from /home/bcumming/software/github/arbor/test/unit-modcc/test_parser.cpp:5:
/home/bcumming/software/github/arbor/test/unit-modcc/test_parser.cpp: In member function ‘virtual void Parser_parse_conductance_Test::TestBody()’:
/home/bcumming/software/github/arbor/test/unit-modcc/test_parser.cpp:167:37: error: ‘ionKind’ has not been declared
         EXPECT_EQ(s->ion_channel(), ionKind::Na);

```

--- a/scripts/travis/build.sh
+++ b/scripts/travis/build.sh
@@ -76,6 +76,10 @@ progress "C++ unit tests"
 make unit -j4                || error "building unit tests"
 ./bin/unit --gtest_color=no  || error "running unit tests"
 
+progress "C++ modcc unit testss"
+make unit-modcc -j4                || error "building modcc unit tests"
+./bin/unit-modcc --gtest_color=no  || error "running modcc unit tests"
+
 progress "C++ distributed unit tests (local)"
 make unit-local -j4          || error "building local distributed unit tests"
 ./bin/unit-local             || error "running local distributed unit tests"

--- a/test/unit-modcc/test_parser.cpp
+++ b/test/unit-modcc/test_parser.cpp
@@ -164,13 +164,13 @@ TEST(Parser, parse_conductance) {
 
     EXPECT_TRUE(check_parse(s, &Parser::parse_conductance, "CONDUCTANCE g USEION na"));
     if (s) {
-        EXPECT_EQ(s->ion_channel(), ionKind::Na);
+        EXPECT_EQ(s->ion_channel(), "na");
         EXPECT_EQ(s->name(), "g");
     }
 
     EXPECT_TRUE(check_parse(s, &Parser::parse_conductance, "CONDUCTANCE gnda"));
     if (s) {
-        EXPECT_EQ(s->ion_channel(), ionKind::nonspecific);
+        EXPECT_EQ(s->ion_channel(), "");
         EXPECT_EQ(s->name(), "gnda");
     }
 }

